### ISSUE TITLE
Nonsensical addition of role in require-valid-alt-text

### ISSUE THREAD

**Issue by ZoeBijl** [ember-template-lint/ember-template-lint#2190]
The [require-valid-alt-text rule](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/require-valid-alt-text.md) states (emphasis mine):

> An `<img>` must have the `alt` attribute. It must have either meaningful text, or be an empty string. If it is an empty string, the `<img>` element tag _must also have `role="presentation"` or `role="none"`_.

That last part makes no sense to me. Why is this rule forcing people to add roles that aren’t necessary? The image is already exposed as decoration/irrelevant, why remove the semantics? That doesn’t add anything I don’t think.

Am I missing something?

---

In addition, `<img src="img.jpg" alt="" role="presentation">` is invalid HTML according to the [W3C Nu Validator](https://validator.w3.org/nu/#textarea).

[Related twitter thread](https://twitter.com/ZoeBijl/status/1460575553065521159)

**Comment by Windvis** [comment]
@ZoeBijl This issue was brought up in the [Discord](https://discord.com/channels/480462759797063690/480523441082990612/831862406439108629) and it was indeed concluded that changes needed to be made to this linting rule. 

I guess this was just lost out of sight and a PR needs to be made :+1:.

**Comment by bmish** [comment]
It might help to tag the original author of that requirement here and link to the PR where it was implemented? That could help us gather some context.

But yes, it sounds like we can treat this as a bug and someone can adjust/improve the rule.

**Comment by MelSumner** [comment]
it was originally a misinterpretation of the specification but it's the only way we have to lint alt text on images, so I think we should make it optional rather than remove it. There's no way to determine intent otherwise. 

Examples: 

We know that this author intends to provide a descriptive text: 
`<img src="foo.png" alt="a red foo with big yellow eyes" />` 

We know that this author does not intend to provide descriptive text: 
`<img src="foo.png" alt="" role="presentation" />

We don't know if this author forgot to add text, doesn't intend to add text, or didn't know they were supposed to add text: 
`<img src="foo.png" alt="" />`

**Comment by Windvis** [comment]
> There's no way to determine intent otherwise.

@MelSumner I don't think I agree with that. Adding an empty alt text makes it pretty clear to me that the image is decorative. If you can't trust spec compliant code to determine intent then I don't know what code you can trust :smile:. Just because something can be valid by accident (because your editor generates it like that) doesn't mean we should change the rules.

> We know that this author does not intend to provide descriptive text:
`<img src="foo.png" alt="" role="presentation" />`

But this isn't valid according to the spec so suggesting that (and enforcing it in the current implementation) isn't a good solution in my opinion.

I still think that this should be considered a bug and that the current implementation should be adjusted so users aren't forced to add a role attribute when it isn't needed (or not allowed even).
